odoo/tools/translate.py

```diff
@@ -888,6 +889,45 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_spreadsheet_terms(fileobj, keywords, comment_tags, options):
+    """Babel message extractor for spreadsheet data files.
+
+    :param fileobj: the file-like object the messages should be extracted from
+    :param keywords: a list of keywords (i.e. function names) that should
+                     be recognized as translation functions
+    :param comment_tags: a list of translator tags to search for and
+                         include in the results
+    :param options: a dictionary of additional options (optional)
+    :return: an iterator over ``(lineno, funcname, message, comments)``
+             tuples
+    :rtype: Iterable
+    """
+    terms = []
+    data = json.load(fileobj)
+    for sheet in data.get('sheets', []):
+        for cell in sheet['cells'].values():
+            content = cell.get('content', '')
+            terms += re.findall(r'_t\("(.+?)"\)', content, re.IGNORECASE)
```

Comment:
```suggestion
            terms += re.findall(r'''_t\((["'])(.+?)\1\)''', content, re.IGNORECASE)
```
Unless the spreadsheet system itself only allows double quotes, all allowable quotes should be handled properly by the extractor.

It also seems like it'd be broken if a non-formula contains / quotes `_t(...)`, or if a formula cell contains an `_)t` *inside* a string literal for some reason.

And why the ignorecase? Is `_T` a valid translation function?

Reply:
Actually _t("...") inside a string literal can't happen (or would be invalid as a spreadsheet string literal anyway). The quotes would need to be escaped in the formula: `'="_t(\\"coucou\\")"'` and it wouldn't match.
Do you think we still should be protected against gugusse that would commit invalid spreadsheet data in it's code ?

Reply:
> But I don't know how to handle _t( inside string literals though, I'm only a level 0 regexp apprentice

I don't think you can do it using regexes (at least not Python's), you have to parse the content for real.

Given spreadsheet seems to use a pretty restricted form of expressions, you might be able to just use Python's [tokenize](https://docs.python.org/3/library/tokenize.html?highlight=tokenize#module-tokenize) module.